pkg/migrations/mls/20240411200242_init-identity.up.sql

```diff
@@ -0,0 +1,28 @@
+SET statement_timeout = 0;
+
+--bun:split
+
+CREATE TABLE inbox_log (
+    sequence_id BIGSERIAL PRIMARY KEY,
+    inbox_id TEXT NOT NULL,
```

Comment:
One performance optimization we _could_ do here:

We could shrink down some indexes if we had an `inbox` table, and just used a foreign key. `inbox_id`s are 64 bytes and a BIGSERIAL field is 8.

Reply:
Would the `inbox` table just have `id` and `inbox_id` columns?

I'm not sure that would shrink the size - indexes store a hash of the column, not the raw column, right?

Reply:
All queries will need to resolve the inbox log entry starting from the `inbox_id` - so I think that just moves the same index onto a different table, and writes will need to update both tables regardless? I guess we could consider using a [HASH index](https://www.postgresql.org/docs/current/hash-intro.html#:~:text=Hash%20indexes%20support%20only%20single,not%20the%20actual%20column%20value.) instead of a BTREE index, assuming we have the [right postgres version](https://hakibenita.com/postgresql-hash-index#hash-index-in-postgresql)

Reply:
I guess one thing to be careful about - the first link says `As a result of the overflow cases, we can say that hash indexes are most suitable for unique, nearly unique data or data with a low number of rows per hash bucket`. I suppose most inbox_id's will have a low number of rows, but I'm unsure what the limit of 'low' is

Reply:
> so I think that just moves the same index onto a different table, and writes will need to update both tables regardless?

The main advantage is that you only need to store/index it once, instead of storing/indexing the full size per update

Reply:
Is this a common pattern? I'm not sure whether that would be more performant than what postgres does internally with indices (btree or hash, depending on what we want to do). Are we concerned about size from the perspective of latency, or is it intrinsically about size? For latency, note that having an extra table would require an extra JOIN on every read and write.

Reply:
I guess another way of looking at it - do we prefer our data normalized, or denormalized, with the classic tradeoffs involved? A separate table would be more normalized, having it built into one table is more denormalized

Reply:
If we had an old-school DBA here they'd say we should normalize everything we can and that's how SQL databases are designed to work. 

These days the databases are so powerful that a lot of apps optimize for simplicity and denormalize more...and it's usually OK. There's a performance cost, but not enough to cause problems until you get to a really large scale.

Reply:
Summarizing an offline conversation:

We've decided to leave the schema as-is. We'll trade larger index sizes for simpler read queries, which makes sense given that the data isn't duplicated that often (most users will have relatively few identity updates)